### Robert Williams IS362 Project 3 


pwd finding the local of the file

In [22]:
pwd

'/Users/bobby/Desktop/Project3'

The assignment involves combining information from a number of SQL tables, then displaying the information in a pandas DataFrame.

In [23]:
import  pandas as pd
import sqlite3
from sqlalchemy import create_engine

The Project 3 assignment is to create a pandas DataFrame that displays Customers’ Last Name and First Name, and each customer’s purchased Track names and Album Titles. The information should be sorted by Customer LastName then Customer FirstName. 

In [24]:
con = sqlite3.connect('/Users/bobby/Desktop/Project3/Chinook_Sqlite.sqlite')
tracks = pd.read_sql_query("SELECT * from Track", con=con)

In [25]:
sql = '''SELECT c.LastName, c.FirstName, t.Name, a.Title
         FROM Customer as c
         LEFT JOIN Invoice as i
         ON c.CustomerId = i.CustomerId
         LEFT JOIN InvoiceLine as il
         ON i.InvoiceId = il.InvoiceId
         LEFT JOIN Track as t
         ON il.TrackId = t.TrackId
         LEFT JOIN Album as a
         ON t.AlbumId = a.AlbumId
         ORDER BY c.LastName, c.FirstName'''

con = sqlite3.connect("/Users/bobby/Desktop/Project3/Chinook_Sqlite.sqlite")
pd.read_sql_query(sql , con=con)


LastName FirstName                                               Name  \
0        Almeida   Roberto                            Right Next Door to Hell   
1        Almeida   Roberto                                     In The Evening   
2        Almeida   Roberto                                   Fool In The Rain   
3        Almeida   Roberto  Saudade Dos Aviões Da Panair (Conversando No Bar)   
4        Almeida   Roberto                             Caso Você Queira Saber   
...          ...       ...                                                ...   
2235  Zimmermann      Fynn                                    Nothin' To Lose   
2236  Zimmermann      Fynn                         Since I've Been Loving You   
2237  Zimmermann      Fynn                                Going To California   
2238  Zimmermann      Fynn                                 We're Gonna Groove   
2239  Zimmermann      Fynn                                         Ozone Baby   

                             Title  
0              Use Your Illusion I  
1          In Through The Out Door  
2          In Through The Out Door  
3                            Minas  
4                            Minas  
...                            ...  
2235              Unplugged [Live]  
2236  BBC Sessions [Disc 2] [Live]  
2237  BBC Sessions [Disc 2] [Live]  
2238                          Coda  
2239                          Coda  

[2240 rows x 4 columns]

aftering connecting to Chinook_Sqlite file there are about 2240 Rows x 4 columns